### ISSUE TITLE
feat(tabs): [ACCESSIBILITY] possibility to have mat-tab-button instead of mat-tab-link

### ISSUE THREAD

**Issue by GregoireLgt** [angular/components#26182]
### For accessibility purposes,

Is it possible to have the mat-tab-button directive instead of mat-tab-link  when using the Tabs component in general? The mat-tab-link directive adds role="tab" to the link (a tag) in which you put this directive but according to the W3C, it is not recommended to have an a tag with role="tab" attribute as you can see here :

https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html

role="tab" requires a button

To sum up,

Add the possibility to have button mat-tab-button instead of a mat-tab-link

Thanks a lot

### Use Case

There are two corresponding use cases:

- When you want to use mat-tab-group to switch between views within a single route in your application
- When you want to use nav mat-tab-nav-bar for navigating between routes in your application

**Comment by crisbeto** [comment]
Closing since the element type doesn't really matter once a `role` is applied. I'm guessing that the W3C demo is using `button` to distinguish it from the other `div` elements on the page.

**Comment by GregoireLgt** [comment]
> Closing since the element type doesn't really matter once a `role` is applied. I'm guessing that the W3C demo is using `button` to distinguish it from the other `div` elements on the page.

I do not agree with you.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#

"It is recommended to use a <button> element with the role tab for their built-in functional and accessible features instead, as opposed to needing to add them yourself"

**Comment by crisbeto** [comment]
I think that quote is referring to the fact that buttons are tabable by default, but we get around it already by setting a `tabindex`. We already need some custom code for the keyboard navigation anyway so I don't see it as much of an issue. There's also the fact that buttons have some built in behavior that we don't want like submitting forms. We run accessibility assessments against our components and this has never come up as an issue.

**Comment by zarend** [comment]
Hello @GregoireLgt ,

Thank you for reaching out. I'm not sure I understand what this request is asking for, and how it improves accessibility.

> but according to the W3C, it is not recommended to have an a tag with role="tab" attribute

Would you be able to tell me specifically what instructions in the W3C tabs guide that you would like us to follow? that way we can understand how the feature suggestion would affect accessibility.

Best,

Zach
